pandas/core/arrays/datetimes.py

```diff
@@ -357,7 +357,7 @@ def _from_sequence_not_strict(
 
         if inferred_freq is None and freq is not None:
             # this condition precludes `freq_infer`
-            cls._validate_frequency(result, freq, ambiguous=ambiguous)
+            cls._validate_frequency(result, freq, ambiguous=ambiguous, unit=result.unit)
```

Comment:
i have a branch where i added unit=index.unit inside _validate_frequency.  any reason to that here instead of there?

Reply:
sorry which branch?

Reply:
Just a local branch

Reply:
yeah that works too